lib/internal/main/watch_mode.js

```diff
@@ -112,14 +114,16 @@ async function restart() {
 
 (async () => {
   emitExperimentalWarning('Watch mode');
+  const promises = [];
 
   try {
     start();
 
     // eslint-disable-next-line no-unused-vars
     for await (const _ of on(watcher, 'changed')) {
-      await restart();
+      ArrayPrototypePush(promises, restart());
```

Comment:
this changes the behavior of watch mode and intoruces a memory leak.
the loop isnt supposed to end until the process ends.

Reply:
> this changes the behavior of watch mode and intoruces a memory leak. the loop isnt supposed to end until the process ends.

I apologize for the unintended consequences. I'll revert the change promptly to address the issues with watch mode and the memory leak. Thank you for bringing this to my attention.